src/sentry/integrations/msteams/client.py

```diff
@@ -93,14 +89,52 @@ def send_welcome_message(self, conversation_id):
         self.send_message(conversation_id, payload)
 
 
-class OauthMsTeamsClient(ApiClient):
+# MsTeamsClient is used with the access token and service url as arguments to the constructor
+# It will not handle token refreshing
+class MsTeamsClient(MsTeamsAbstractClient):
+    def __init__(self, access_token, service_url):
+        super(MsTeamsClient, self).__init__()
+        self.access_token = access_token
+        self.base_url = service_url.rstrip("/")
+
+
+# MsTeamsIntegrationClient is used with an existing integration object and handles token refreshing
+class MsTeamsIntegrationClient(MsTeamsAbstractClient):
+    def __init__(self, integration):
+        super(MsTeamsIntegrationClient, self).__init__()
+        self.integration = integration
+
+    @property
+    def metadata(self):
+        return self.integration.metadata
+
+    @property
+    def base_url(self):
+        return self.metadata["service_url"].rstrip("/")
+
+    @property
+    def access_token(self):
+        access_token = self.metadata.get("access_token")
+        expires_at = self.metadata.get("expires_at")
+
+        # if the token is expired, refresh it and save  it
+        if expires_at <= int(time.time()):
+            token_data = get_token_data()
+            access_token = token_data["access_token"]
+            self.metadata.update(token_data)
+            self.integration.save()
+        return access_token
+
+
+# OAuthMsTeamsClient is used only for the exchanging the token
+class OAuthMsTeamsClient(ApiClient):
     base_url = "https://login.microsoftonline.com/botframework.com"
     integration_name = "msteams_oauth"
```

Comment:
Do we really need the `integration_name` to be different? For transactions I feel like we could add a tag or something. Might be a little more work upfront to figure out the best way to do that but it could mean other integrations to add more data to the transaction (like how slack has the `slack_error` and `ok`)

Reply:
@MeredithAnya We do this with other integrations but I don't think it adds value, I'll keep it as `msteams`.